### PR TITLE
RR-107 - Basic config and types for calling curious-api

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR adds the basic types and config to support being able to call the curious API

### CODE REVIEW

File: server/@types/curiousApiClient/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+declare module 'curiousApiClient' {
+  import { components } from '../curiousApi'
+
+  export type LearnerEducation = components['schemas']['LearnerEducationDTO']
+  export type LearnerNeurodivergence = components['schemas']['LearnerNeurodivergenceDTO']
```
Comment (nathanrussell-moj-digital) [inline server/@types/curiousApiClient/index.d.ts:5]: There are other types in the Curious API schema that we will need as we get into more stories that integrate with Curious; but this is sufficient for now.  (In fact all we really need for this story is `LearnerNeurodivergenceDTO` but I also added `LearnerEducationDTO` as we will need that for subsequent stories)

File: README.md [hunk 2]
```diff
@@ -114,6 +117,19 @@ And then, to build the assets and start the app with nodemon:
...
+* `ROLE_PRISONER_SEARCH` - to be able to call the Prisoner Search API
+* `ROLE_CURIOUS_API` - to be able to call the Curious API
+
+### User Roles
+One the UI is running users will need to authenticate with `hmpps-auth` using a valid DPS username. The DPS user needs to have the followings roles
+dependent on the access/functionality required:
+
+* `ROLE_EDUCATION_WORK_PLAN_VIEWER` - required to be able to perform read only actions
+* `ROLE_EDUCATION_WORK_PLAN_EDITOR` - required to be able to perform actions that edit/create data
+
```
Comment (mattwills-moj-digital) [inline README.md:132]: 🙌

File: helm_deploy/values-prod.yaml [hunk 3]
```diff
@@ -11,6 +11,7 @@ generic-service:
     TOKEN_VERIFICATION_API_URL: "https://token-verification-api.prison.service.justice.gov.uk"
     EDUCATION_AND_WORK_PLAN_API_URL: "https://learningandworkprogress-api.hmpps.service.justice.gov.uk"
     PRISONER_SEARCH_API_URL: "https://prisoner-offender-search.prison.service.justice.gov.uk"
+    CURIOUS_API_URL: "https://liveservices.sequation.net/sequation-virtual-campus2-api/"
```
Comment (mattwills-moj-digital) [inline helm_deploy/values-prod.yaml:14]: Does this mean VC2 and Curious are the same thing?

**Review by mattwills-moj-digital — APPROVED** [review]
ATAB

File: helm_deploy/values-prod.yaml (hunk 3 above)
Comment (nathanrussell-moj-digital) [reply helm_deploy/values-prod.yaml:14]: Yes; as I understand it they are one and the same thing

**Pull request merged**